src/button-dropdown/category-elements/expandable-category-element.tsx

```diff
@@ -37,6 +38,12 @@ const ExpandableCategoryElement = ({
     }
   }, [expanded, highlighted]);
 
+  useEffect(() => {
```

Comment:
Why do we need three useEffect calls?

I'd expect that to be performed only once in the component's root. Something like:

```
function InternalButtonDropdown() {
  // ...
  
  useEffect(() => {
    if (isOpen) {
      // focus first item
    }
  }, [isOpen])
  
  // ...
}
```

The "focus first item" part can be done in two ways:

1. Focus using selector: `buttonDropdownRef.current?.querySelector('.' + styles.item)?.focus()`.
2. Focus using reference:
```
function InternalButtonDropdown() {
  const firstItemRef = useRef(null)
  
  useEffect(() => {
    if (isOpen) {
      firstItemRef.current?.focus()
    }
  }, [isOpen])
  
  // ... <ItemsList ... firstItemRef={firstItemRef} />
}

function Item(props) {
  return <div ... ref={props.isFirst ? props.firstItemRef : undefined} />
}
```

What do you think?

Reply:
The `useButtonDropdown` can probably return an extra property -- `isFirstItem` which, when taken an item would return true if and only if the item is the first visible item in the list. That can be used to assign the ref to it.

However, I don't mind using a selector as well.

Reply:
Example using query selector: https://github.com/cloudscape-design/components/blob/dda400a1aec2bae02b199be0152423c9be4ac786/src/calendar/internal.tsx#L112

Reply:
thanks for the suggestions andrei, I'll try to implement the query selector solution. 
the current solution is failing integ tests anyway so I have to change it